applications/ConstitutiveLawsApplication/custom_constitutive/yield_surfaces/rankine_yield_surface.h

```diff
@@ -120,15 +122,13 @@ class RankineYieldSurface
         ConstitutiveLaw::Parameters& rValues
         )
     {
-        double I1, J2, J3, lode_angle;
-        array_1d<double, VoigtSize> deviator = ZeroVector(VoigtSize);
-
-        AdvancedConstitutiveLawUtilities<VoigtSize>::CalculateI1Invariant(rPredictiveStressVector, I1);
-        AdvancedConstitutiveLawUtilities<VoigtSize>::CalculateJ2Invariant(rPredictiveStressVector, I1, deviator, J2);
-        AdvancedConstitutiveLawUtilities<VoigtSize>::CalculateJ3Invariant(deviator, J3);
-        AdvancedConstitutiveLawUtilities<VoigtSize>::CalculateLodeAngle(J2, J3, lode_angle);
-
-        rEquivalentStress = 2.0 * std::sqrt(3 * J2) / 3.0 * std::cos(lode_angle + Globals::Pi / 6.0) + I1 / 3.0;
+        array_1d<double, Dimension> principal_stress_vector = ZeroVector(Dimension);
+        AdvancedConstitutiveLawUtilities<VoigtSize>::CalculatePrincipalStresses(principal_stress_vector, rPredictiveStressVector);
+        // The rEquivalentStress is the maximum principal stress
+        if (Dimension == 3) 
```

Comment:
To mention once we move to C++17 we can add here a constexpr so the if is done in compilation time
```suggestion
        if (Dimension == 3)  // TODO: Add constexpr with C++17
```